File: frontend/src/components/SetPriceAlertRequest.js
import axios from 'axios';

const setPriceAlertRequest = async (userId, stockSymbol, targetPrice, alertType) => {
  try {
    const response = await axios.post('http://localhost:5244/api/stocks/price-alert', {
      userId: userId,
      stockSymbol: stockSymbol,
      targetPrice: targetPrice,
      alertType: alertType,
    });

    return response.data;
  } catch (error) {
    console.error("Error setting price alert:", error);
    throw error;
  }
};

export default setPriceAlertRequest;

File: backend/API/Contracts/SetPriceAlertRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Threading.Tasks;
using Core.Data.Entity;

namespace API.Contracts
{
    public class SetPriceAlertRequest
    {
        [DefaultValue("3aa42229-1c0f-4630-8c1a-db879ecd0427")]
        public Guid UserId { get; set; }
        [DefaultValue("AAPL")]
        public string StockSymbol { get; set; } = string.Empty;
        [DefaultValue(100.00)]
        public decimal TargetPrice { get; set; }
        [DefaultValue("Rise")]
        public AlertType AlertType { get; set; }


    }
}